ocaml/testsuite/tests/typing-higher-jkinds/equalities.ml

```diff
@@ -0,0 +1,110 @@
+(* TEST
+ flags = "-extension layouts_alpha";
+ expect;
+*)
+
+type t : value => value
+
+[%%expect {|
+type t : value => value
+|}]
+
+type ('a : (value, (value) => value) => value) t
+
+[%%expect {|
+type ('a : (value, value => value) => value) t
+|}]
+
+module M : sig
+  type a : value => value
+end = struct
+  type a : value => value
+end
+
+[%%expect {|
+module M : sig type a : value => value end
+|}]
+
+module M : sig
+  type a : value => value
+end = struct
+  type a : value
+end
+
+[%%expect {|
+Lines 3-5, characters 6-3:
+3 | ......struct
+4 |   type a : value
+5 | end
+Error: Signature mismatch:
+       Modules do not match:
+         sig type a : value end
+       is not included in
+         sig type a : value => value end
+       Type declarations do not match:
+         type a : value
+       is not included in
+         type a : value => value
+       The layout of the first is value, because
+         of the definition of a at line 4, characters 2-16.
+       But the layout of the first must be a sublayout of ((value) => value) (...??)
```

Comment:
What is causing the `??` part? Probably worth fixing that sooner rather than later.

Reply:
History tracking not implemented for higher jkinds. Fixed a few PRs later down the line.